src/event.js

```diff
@@ -547,16 +547,13 @@ function leverageNative( el, type, isSetup ) {
 				// If this is an inner synthetic event for an event with a bubbling surrogate
 				// (focus or blur), assume that the surrogate already propagated from triggering
 				// the native event and prevent that from happening again here.
-				// This technically gets the ordering wrong w.r.t. to `.trigger()` (in which the
-				// bubbling surrogate propagates *after* the non-bubbling base), but that seems
-				// less bad than duplication.
 				} else if ( ( jQuery.event.special[ type ] || {} ).delegateType ) {
 					event.stopPropagation();
 				}
 
 			// If this is a native event triggered above, everything is now in order
 			// Fire an inner synthetic event with the original arguments
-			} else if ( saved ) {
+			} else if ( saved.length ) {
```

Comment:
I wonder if I should also replace the upper `if ( !saved ) {` with a `!saved.length` check. I haven't yet found a way to confuse that other `if` since it's already inside an `isTrigger` block but before 3.7.0 both were checking for `saved.length` instead of just `saved`.